soundfile.py

```diff
@@ -986,7 +986,9 @@ def _open(self, file, mode_int, closefd):
             file_ptr = _snd.sf_open(file, mode_int, self._info)
         elif isinstance(file, int):
             file_ptr = _snd.sf_open_fd(file, mode_int, self._info, closefd)
-        elif all(hasattr(file, a) for a in ('seek', 'read', 'write', 'tell')):
+        elif ( hasattr(file, 'seek') and hasattr(file, 'tell') and
+               (hasattr(file, 'read') or mode_int == _snd.SFM_WRITE) and
```

Comment:
While you're at it, you could add "`or hasattr(file, 'readinto')`".
We had support for this already, but we never checked for it ...